custom_components/tahoma/__init__.py

```diff
@@ -132,10 +134,24 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
         "update_listener": entry.add_update_listener(update_listener),
     }
 
+    def has_state(states, names):
+        return any(state.qualified_name in names for state in states)
+
     for device in tahoma_coordinator.data.values():
         platform = TAHOMA_TYPES.get(device.widget) or TAHOMA_TYPES.get(device.ui_class)
         if platform:
             entities[platform].append(device)
+            if device.definition.states and has_state(
+                device.definition.states,
+                [
+                    "core:BatteryState",
+                    "core:BatteryLevelState",
+                    "core:SensorDefectState",
+                ],
+            ):
+                new_device = deepcopy(device)
```

Comment:
```suggestion
                battery_entity = deepcopy(device)
```

Should we rename this variable, since it is always used for battery entities?